### PR TITLE
Refactor shuffle method to handle invalid columns

### PR DESCRIPTION
**Pull request by alex-rakowski**
closes https://github.com/dask/dask/issues/11174

Handles shuffling on columns that do not exist. Raises a KeyError.

### CODE REVIEW

File: dask_expr/_collection.py [hunk 1]
```diff
@@ -896,8 +896,24 @@ def shuffle(
...
+            if pd.api.types.is_list_like(on) and not is_dask_collection(on):
+                on = list(on)
+            elif isinstance(on, str):
+                on = [on]
+            bad_cols = [
+                index_col
+                for index_col in on
+                if (index_col not in self.columns) and (index_col != self.index.name)
+            ]
+            if len(bad_cols) == 1:
```
Comment (phofl) [inline dask_expr/_collection.py:909]: I think you can reformat this into a single statement, like:

f"Cannot shuffle on {bad_cols}, column(s) not in dataframe to shuffle"

**Comment by alex-rakowski** [comment]
Only `dask_expr/tests/test_groupby.py::test_groupby_index_array` fails which is expected.

File: dask_expr/tests/test_shuffle.py [hunk 2]
```diff
@@ -130,6 +130,39 @@ def test_task_shuffle_index(npartitions, max_branch, pdf):
...
+        KeyError,
+        match="Cannot shuffle on",
+    ) as execinfo:
+        df.shuffle(["x", "z"])
+    assert "z" in str(execinfo.value)
+    assert "x" not in str(execinfo.value)
+
+
+def test_shuffle_list_column_not_in_dataframe(df):
+    # all cols in list are not in dataframe
```
Comment (phofl) [inline dask_expr/tests/test_shuffle.py:155]: We don't need comments here (only for very complicated tests, the test name should be verbose enough to tell us what's going on here)

File: dask_expr/tests/test_shuffle.py [hunk 3]
```diff
@@ -130,6 +130,39 @@ def test_task_shuffle_index(npartitions, max_branch, pdf):
     assert sorted(df3.compute().values) == list(range(20))
 
 
+def test_shuffle_str_column_not_in_dataframe(df):
+    # ddf = from_pandas(pdf, npartitions=10)
```
Comment (phofl) [inline dask_expr/tests/test_shuffle.py:134]: This can be removed?

File: dask_expr/_collection.py [hunk 4]
```diff
@@ -912,7 +924,6 @@ def shuffle(
                     f"p2p requires all column names to be str, found: {unsupported}",
                 )
 
-        # Returned shuffled result
```
Comment (phofl) [inline dask_expr/_collection.py:915]: Can you revert this change?

**Review by phofl — CHANGES_REQUESTED** [review]

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx
